blockchain/v0/pool.go

```diff
@@ -163,26 +174,22 @@ func (pool *BlockPool) GetStatus() (height int64, numPending int32, lenRequester
 }
 
 // IsCaughtUp returns true if this node is caught up, false - otherwise.
-// TODO: relax conditions, prevent abuse.
 func (pool *BlockPool) IsCaughtUp() bool {
 	pool.mtx.Lock()
 	defer pool.mtx.Unlock()
 
+	if time.Since(pool.lastAdvance) > syncTimeout {
+		pool.Logger.Error(fmt.Sprintf("No progress since last advance: %v", pool.lastAdvance))
+		return true
+	}
```

Comment:
I don't think this should be in `IsCaughtUp` -- we're not caught up, the sync has timed out. Maybe just add this check explicitly during the switch to consensus ticker event.